client/src/templates/Challenges/redux/current-challenge-saga.js

```diff
@@ -16,14 +18,20 @@ import { post } from '../../../utils/ajax';
 import { randomCompliment } from '../utils/get-words';
 import { updateSuccessMessage } from './';
 
-function* currentChallengeSaga({ payload }) {
+export const CURRENT_CHALLENGE_KEY = 'currentChallengeId';
+
+export function* currentChallengeSaga({ payload: id }) {
   const isSignedIn = yield select(isSignedInSelector);
   const currentChallengeId = yield select(currentChallengeIdSelector);
-  if (isSignedIn && payload !== currentChallengeId) {
+  // TODO: should the server update take place on challenge completion instead?
```

Comment:
Do we need it?

Reply:
Not sure, that's why I left the TODO.  @raisedadead?

Reply:
Yes, this is the note that I was referring to our design of the system earlier. We intended to update the challenge update only on successful completion. The challenge navigation should have been kept different from this. 

Anyway, since we are taking a new direction, lets remove this note.